python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -533,6 +535,45 @@ def selected_asset_keys(self) -> AbstractSet[AssetKey]:
             return set()
         return self.assets_def.keys
 
+    @public
+    @property
+    def has_asset_checks_def(self) -> bool:
+        """Return a boolean indicating the presence of a backing AssetChecksDefinition
+        for the current execution.
+
+        Returns:
+            bool: True if there is a backing AssetChecksDefinition for the current execution, otherwise False.
+        """
+        return self.job_def.asset_layer.asset_checks_def_for_node(self.node_handle) is not None
+
+    @public
+    @property
+    def asset_checks_def(self) -> AssetChecksDefinition:
+        """The backing AssetChecksDefinition for what is currently executing, errors if not
+        available.
+
+        Returns:
+            AssetChecksDefinition.
+        """
+        asset_checks_def = self.job_def.asset_layer.asset_checks_def_for_node(self.node_handle)
+        if asset_checks_def is None:
+            raise DagsterInvalidPropertyError(
+                f"Op '{self.op.name}' does not have an asset checks definition."
+            )
+
+        return asset_checks_def
+
+    @public
+    @property
+    def selected_asset_check_handles(self) -> AbstractSet[Tuple[AssetKey, str]]:
```

Comment:
same question as https://dagsterlabs.slack.com/archives/C05KFG9TETB/p1695240441466179 cc @schrockn - we could return specs here instead, or just call it `selected_asset_checks`...

Reply:
I think we should call it `selected_asset_check_handles` and return a collection of `AssetCheckHandle`s. If we don't think "handle" is the right name, then worth discussing, but I don't think expecting users to access the attributes with positional indexing is a good experience.